contracts/chain-adapters/WorldChain_Adapter.sol

```diff
@@ -85,15 +88,17 @@ contract WorldChain_Adapter is CrossDomainEnabled, AdapterInterface, CircleCCTPA
         if (l1Token == address(L1_WETH)) {
             L1_WETH.withdraw(amount);
             L1_STANDARD_BRIDGE.depositETHTo{ value: amount }(to, L2_GAS_LIMIT, "");
-        }
-        // Check if this token is USDC, which requires a custom bridge via CCTP.
-        else if (_isCCTPEnabled() && l1Token == address(usdcToken)) {
-            _transferUsdc(to, amount);
+        } else if (l1Token == address(usdcToken)) {
+            if (_isCCTPEnabled()) {
+                _transferUsdc(to, amount);
+            } else {
+                // Use WorldChain's OP USDC bridge to received bridged USDC on L2.
+                IERC20(l1Token).safeIncreaseAllowance(address(L1_OP_USDC_BRIDGE), amount);
+                L1_OP_USDC_BRIDGE.sendMessage(to, amount, L2_GAS_LIMIT);
+            }
         } else {
```

Comment:
There's talk of wanting to reuse this on future adapters so we don't need audit on those. Are we OK w/ a copy/paste of this contract on future OP stack chains? That's what we have done previously for Mode, Lisk, Redstone & Zora.

I'm tempted to factor it out somehow though, such that it's more modular and we can more clearly identify it as pre-audited code.

Reply:
Can we not put this in the ovm adapter?

Reply:
Yeah; we don't have an OVM adapter but I agree it'd be best to implement one - then we get maximum reuse here.

Reply:
But is this change required for any other chains today?

Reply:
I'm not aware of any that don't use L1StandardBridge for bridged USDC, so this is a new requirement. I think we should anticipate it popping up on subsequent new deployments, though.

Reply:
Understood so let's just not rename this contract to `OpStack_Adapter` or something more generic for the purpose of keeping this PR diff small